### DISCUSSION TITLE
Disable auto merge when main branch is red

### DISCUSSION THREAD

**Discussion by gbleu** [renovatebot/renovate#26616] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.128.7

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### What is your question?

We have extra checks on master, like end to end tests on a shared environment, sometimes master can be red for couple days  but merge request pipelines still passing.

In this scenario we would like to postprone dependency updates, is that already achievable ? Would this interest other people ?

**Comment by rarkins** [comment]
That's an interesting scenario. So you have commit A on branch "main", and then Renovate creates commit B on branch "renovate/something" which has parent commit A. A fails checks, B passes checks. But you don't want to automerge B into A?

BTW these days I normally advise people to use the platform-native automerge when it's available, such as on GitLab. But you are using Renovate's own automerge, not the platform one? Because if you already solved this for GitLab automerge then you wouldn't need Renovate to solve it again.

**Comment by gbleu** [comment]
Yes

Definitly using platform-native automerge but since B passes checks it ends up beeing merged.

We could add a check to fail the MR pipeline when the master pipeline is red but it comes with a lot of drawbacks:
- we might need to merge in order to fix master
- we still need to merge certain types of changes that does not need to get deployed, like db queries
- we do not require fast-fowrard merge (~100 merge/day), a merge request without conflict with a pipeline passing from last week (master was green at that time) can still be merged

**Comment by gbleu** [comment]
@rarkins let's see the the other way around, in wich case would we like to enable auto merge when the target branch is already broken ?

**Reply by rarkins** [reply]
If we're being snarky now, why would want to have such bad tests on your MRs that you can't trust them even when they pass?

I don't think we'll add this feature. Suggest you disable automerge altogether

**Reply by gbleu** [reply]
Sorry, was not supposed to be a joke or sommething, the question is not if the MRs tests can be trusted or not, it's just extra checks on main branch like acceptance/capacity tests that are too slow/costy to run on every single MR.

When the extra checks fails, the priority is to come back to a stable state and delaying dependency updates seems the right thing to do to prevent things getting worse or deploying a big batch of updates at once.

**Reply by rarkins** [reply]
Renovate enables automerge at MR creation time, meaning that the platform takes care of automerging later once all tests have passed. At the time MR creation, there might still even be pipelines in progress for the base branch commit.

So you'd need something like:
- Only enable MR automerge at time of MR creation if the base commit has all green statuses
- Enable MR automerge on later runs if base branch commit is green and MR commit is yellow or green
- Somehow avoid enabling automerge if someone has manually disabled it?

**Reply by gbleu** [reply]
How is working `automergeSchedule`, only opening MR during the time range ?

**Reply by rarkins** [reply]
I don't think automergeSchedule is compatible with platformAutomerge because Renovate can't control _when_ the platform decides to automerge.
